testing/deprecated_test.py

```diff
@@ -20,6 +20,54 @@ def test_external_plugins_integrated(pytester: Pytester, plugin) -> None:
         pytester.parseconfig("-p", plugin)
 
 
+def test_hookspec_via_function_attributes_are_deprecated():
+    from _pytest.config import PytestPluginManager
+
+    pm = PytestPluginManager()
+
+    class DeprecatedHookMarkerSpec:
+        def pytest_bad_hook(self):
+            pass
+
+        pytest_bad_hook.historic = True  # type: ignore[attr-defined]
```

Comment:
```suggestion
        pytest_bad_hook.historic = False  # type: ignore[attr-defined]
```

This way we will catch the issue I commented in `_get_legacy_hook_marks`.